go/vt/vtgate/engine/insert.go

```diff
@@ -40,322 +36,78 @@ import (
 
 var _ Primitive = (*Insert)(nil)
 
-type (
-	// Insert represents the instructions to perform an insert operation.
-	Insert struct {
-		// Opcode is the execution opcode.
-		Opcode InsertOpcode
+// Insert represents the instructions to perform an insert operation.
+type Insert struct {
+	*InsertCommon
 
-		// Ignore is for INSERT IGNORE and INSERT...ON DUPLICATE KEY constructs
-		// for sharded cases.
-		Ignore bool
+	// Query specifies the query to be executed.
+	// For InsertSharded plans, this value is unused,
+	// and Prefix, Mid and Suffix are used instead.
+	Query string
 
-		// Keyspace specifies the keyspace to send the query to.
-		Keyspace *vindexes.Keyspace
+	// VindexValues specifies values for all the vindex columns.
+	// This is a three-dimensional data structure:
+	// Insert.Values[i] represents the values to be inserted for the i'th colvindex (i < len(Insert.Table.ColumnVindexes))
+	// Insert.Values[i].Values[j] represents values for the j'th column of the given colVindex (j < len(colVindex[i].Columns)
+	// Insert.Values[i].Values[j].Values[k] represents the value pulled from row k for that column: (k < len(ins.rows))
+	VindexValues [][][]evalengine.Expr
 
-		// Query specifies the query to be executed.
-		// For InsertSharded plans, this value is unused,
-		// and Prefix, Mid and Suffix are used instead.
-		Query string
+	// Prefix, Mid and Suffix are for sharded insert plans.
+	Prefix string
```

Comment:
both Insert and InsertSelect has Prefix and Suffix. That shouldn't move to the InsertCommon struct?